File: android/src/main/kotlin/com/flutter/kbz/pay/flutter_kbz_pay/SHA.kt
package com.flutter.kbz.pay.flutter_kbz_pay

import java.io.UnsupportedEncodingException
import java.nio.charset.Charset
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException

object SHA {
    fun getSHA256Str(str: String): String {
        val messageDigest: MessageDigest
        var encodeStr = ""
        try {
            messageDigest = MessageDigest.getInstance("SHA-256")
            messageDigest.update(str.toByteArray(Charset.forName("UTF-8")))
            encodeStr = byte2Hex(messageDigest.digest())
        } catch (e: NoSuchAlgorithmException) {
            e.printStackTrace()
        } catch (e: UnsupportedEncodingException) {
            e.printStackTrace()
        }
        return encodeStr
    }

    private fun byte2Hex(bytes: ByteArray): String {
        val stringBuffer = StringBuffer()
        var temp: String? = null
        for (i in bytes.indices) {
            temp = Integer.toHexString(bytes[i].toInt() and 0xFF)
            if (temp.length == 1) {
                // 1得到一位的进行补0操作
                stringBuffer.append("0")
            }
            stringBuffer.append(temp)
        }
        return stringBuffer.toString()
    }
}

File: example/lib/demo/SHA.dart
import 'dart:convert';
import 'dart:typed_data';

import 'package:crypto/crypto.dart';

class SHA {
  factory SHA() {
    return I;
  }

  SHA._();

  static final SHA I = SHA._();

  String getSHA256Str(String str) {
    var encodeStr = "";
    try {
      final bytes = utf8.encode(str);
      final digest = sha256.convert(bytes);
      encodeStr = _byte2Hex(Uint8List.fromList(digest.bytes));
    } catch (e) {
      print(e);
    }
    print("sha256 sign $encodeStr");
    return encodeStr;
  }

  String _byte2Hex(Uint8List bytes) {
    var stringBuffer = StringBuffer();
    String temp;
    for (var i = 0; i < bytes.length; i++) {
      temp = bytes[i].toRadixString(16);
      if (temp.length == 1) {
        stringBuffer.write("0");
      }
      stringBuffer.write(temp);
    }
    return stringBuffer.toString();
  }
}
